homeassistant/components/shelly/light.py

```diff
@@ -83,7 +83,7 @@ def __init__(self, wrapper: ShellyDeviceWrapper, block: Block) -> None:
         )
         if hasattr(block, "brightness") or hasattr(block, "gain"):
             self._supported_features |= SUPPORT_BRIGHTNESS
-        if hasattr(block, "colorTemp") and not self._color_mode:
+        if hasattr(block, "colorTemp"):
```

Comment:
This change does not work well on Shelly Bulb.
@balloob @chemelli74 it was added after testing. This Bulb has two mode of operation:
1. Color mode
2. White mode

In color mode there is no "color temperature" option
In white mode there is no "rgb colors" option
We already remove the RGB options in white mode, but now it will show a temperature slider in color mode which is not doing anything.

Reply:
If we want to leave the slider than another option is if the user is moving the slider change the bulb mode into "white" mode and if the user is selecting a color from the color palette to switch the bulb into "color" mode. This is how Philips lights work in HA.
In this case the we should also do not remove the "color palette" when the bulb is in white mode.

Reply:
Correct. That's how it should behave. We should always show all sliders and change mode as needed.

Reply:
Good job, bulb will now switch between color and white mode!